builds.go

```diff
@@ -0,0 +1,431 @@
+		Description string `json:"description"`
+		Active      bool   `json:"active"`
+		IsShared    bool   `json:"is_shared"`
+		Name        string `json:"Name"`
+	} `json:"runner, omitempty"`
+	Stage     string     `json:"stage"`
+	StartedAt *time.Time `json:"started_at"`
+	Status    string     `json:"status"`
+	Tag       bool       `json:"tag"`
+	User      *struct {
+		AvatarURL  string     `json:"avatar_url"`
+		BIO        string     `json:"bio"`
+		CreatedAt  *time.Time `json:"created_at"`
+		ID         int        `json:"id"`
+		IsAdmin    bool       `json:"is_admin"`
+		Linkedin   string     `json:"linkedin"`
+		Name       string     `json:"name"`
+		Skype      string     `json:"skype"`
+		State      string     `json:"state"`
+		Twitter    string     `json:"twitter"`
+		Username   string     `json:"username"`
+		WebURL     string     `json:"web_url"`
+		WebsiteURL string     `json:"website_url"`
+	} `json:"user, omitempty"`
+}
+
+// ListProjectBuilds gets a list of builds in a project.
+//
+// The scope of builds to show, one or array of: pending, running,
+// failed, success, canceled; showing all builds if none provided.
+//
+// GitLab API docs:
+// http://docs.gitlab.com/ce/api/builds.html#list-project-builds
+func (s *BuildsService) ListProjectBuilds(pid interface{}, opts *ListBuildsOptions) ([]Build, *Response, error) {
+	project, err := parseID(pid)
+	if err != nil {
+		return nil, nil, err
+	}
+	u := fmt.Sprintf("projects/%s/builds", url.QueryEscape(project))
+
+	req, err := s.client.NewRequest("GET", u, opts)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	builds := []Build{}
+	resp, err := s.client.Do(req, &builds)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return builds, resp, err
+}
+
+// ListCommitBuilds gets a list of builds for specific commit in a
+// project. If the commit SHA is not found, it will respond with 404.
+//
+// GitLab API docs:
+// http://docs.gitlab.com/ce/api/builds.html#list-commit-builds
+func (s *BuildsService) ListCommitBuilds(pid interface{}, sha string, opts *ListBuildsOptions) ([]Build, *Response, error) {
+	project, err := parseID(pid)
+	if err != nil {
+		return nil, nil, err
+	}
+	u := fmt.Sprintf("projects/%s/repository/commits/%s/builds", project, sha)
+
+	req, err := s.client.NewRequest("GET", u, opts)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	builds := []Build{}
+	resp, err := s.client.Do(req, &builds)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return builds, resp, err
+}
+
+// GetSingleBuild gets a single build of a project.
+//
+// GitLab API docs:
+// http://docs.gitlab.com/ce/api/builds.html#get-a-single-build
+func (s *BuildsService) GetSingleBuild(pid interface{}, buildID int) (*Build, *Response, error) {
+	project, err := parseID(pid)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	u := fmt.Sprintf("projects/%s/builds/%d", project, buildID)
+	req, err := s.client.NewRequest("GET", u, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	build := &Build{}
```

Comment:
In the rest of this package this is written as `build := new(Build)` (again for consistency)